targets/TARGET_STM/sleep.c

```diff
@@ -198,9 +199,12 @@ void hal_deepsleep(void)
      *  deep sleep */
     wait_loop(500);
 
+    // Reprogram the timer counter value saved before the deepsleep
     TIM_HandleTypeDef TimMasterHandle;
     TimMasterHandle.Instance = TIM_MST;
     __HAL_TIM_SET_COUNTER(&TimMasterHandle, EnterTimeUS);
+    __HAL_TIM_CLEAR_FLAG(&TimMasterHandle, TIM_FLAG_CC1);
```

Comment:
Call here a function to restore TIM_MST registers

Reply:
done in commit https://github.com/ARMmbed/mbed-os/pull/7352/commits/a838cb21d12473cf5cd29c708bbd8f381f8d37f7